File: app/src/main/java/rusha/x/basket/BasketViewModel.java
package rusha.x.basket;

import androidx.annotation.NonNull;
import androidx.lifecycle.MutableLiveData;

import java.util.Collections;
import java.util.List;

import javax.inject.Inject;

import rusha.x.BaseViewModel;
import rusha.x.SingleLiveEvent;
import rusha.x.product.Product;

public class BasketViewModel extends BaseViewModel {

    @Inject
    public BasketViewModel() {
        super();
    }

    public MutableLiveData<List<BasketItem>> basketItemsLiveData = new MutableLiveData<>(Collections.emptyList());
    public SingleLiveEvent<Product> goToProductDetails = new SingleLiveEvent<>();

    public void onBasketItemClick(@NonNull BasketItem item) {
        goToProductDetails.setValue(item.product);
    }
}

File: app/src/main/java/rusha/x/basket/BasketViewModel.kt
package rusha.x.basket

import androidx.lifecycle.MutableLiveData
import rusha.x.BaseViewModel
import rusha.x.SingleLiveEvent
import rusha.x.product.Product
import javax.inject.Inject

class BasketViewModel @Inject constructor(): BaseViewModel() {
    val basketItemsLiveData = MutableLiveData<List<Basket.Item>>(emptyList())
    val goToProductDetails = SingleLiveEvent<Product>()

    fun onBasketItemClick(item: Basket.Item) {
        goToProductDetails.value = item.product
    }
}